File: OnlinePollingSystem/Ops.Api/Repositories/Entities/Poll.cs
namespace Ops.Api.Repositories.Entities
{
    public class Poll
    {
        public int Id { get; set; }
        public string Question { get; set; }
        public List<PollOption> Options { get; set; }
        public DateTime ExpirationDate { get; set; }
    }

    public class PollOption
    {
        public int Id { get; set; }
        public int PollId { get; set; }
        public string OptionText { get; set; }
        public int VoteCount { get; set; }
    }

    public class Vote
    {
        public int Id { get; set; }
        public int PollId { get; set; }
        public int OptionId { get; set; }
        public string UserIdentifier { get; set; }
    }
}

File: ops-app/src/models/Poll.ts
export interface PollOption {
    id: number;
    pollId: number;
    optionText: string;
    voteCount: number;
}

export interface Poll {
    id: number;
    question: string;
    options: PollOption[];
    hasVoted: boolean;
    votedOptionId?: number;
    expirationDate?: Date;
}

export interface CreatePoll {
    question: string;
    options: string[];
    expirationDate: Date | null;
}